svm/src/transaction_processor.rs

```diff
@@ -620,14 +624,16 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
             if let Some((key, program_to_recompile)) = program_cache.programs_to_recompile.pop() {
                 let effective_epoch = program_cache.latest_root_epoch.saturating_add(1);
                 drop(program_cache);
+                let program_cache_read = self.program_cache.read().unwrap();
```

Comment:
In the line above, we're dropping a write lock and then immediately getting a read lock; why not just get `environments` before dropping the write lock:
```rust
...
let effective_epoch = program_cache.latest_root_epoch.saturating_add(1);
let environments_for_epoch = program_cache.get_environments_for_epoch(effective_epoch);
drop(program_cache);
```
Granted, this would mean we hold the write lock for a bit longer, but it looks like `ProgramCache::get_environments_for_epoch()` is fairly light (up to two conditionals + cloning two `Arc`'s). But, removing the additional read lock is worth something too.

Reply:
This is just to revert it back to the original code, before the SVM related changes were made.

Reply:
> This is just to revert it back to the original code

To confirm, you're saying the goal of this PR is just to get back to original code, and you'd like to make that change and consider my question / suggestion in an alternate PR ? If so, fine with me

Reply:
Yes, being careful about the scope of the PR.

Reply:
Works for me - ignore this comment for now